graspologic/partition/leiden.py

```diff
@@ -0,0 +1,402 @@
+        starting_communities,
+        iterations,
+        resolution,
+        randomness,
+        use_modularity,
+        random_seed,
+        is_weighted,
+        weight_default,
+        check_directed,
+    )
+    graph = _validate_and_build_edge_list(
+        graph, is_weighted, weight_attribute, check_directed, weight_default
+    )
+
+    if iterations < 1:
+        raise ValueError("iterations must be > 1")
+
+    _improved, _modularity, partitions = gn.leiden(
+        edges=graph,
+        starting_communities=starting_communities,
+        resolution=resolution,
+        randomness=randomness,
+        iterations=iterations,
+        use_modularity=use_modularity,
+        seed=random_seed,
+    )
+
+    return partitions
+
+
+class HierarchicalCluster(NamedTuple):
+    node: str
+    cluster: int
+    parent_cluster: Optional[int]
+    level: int
+    is_final_cluster: bool
+
+    @classmethod
+    def from_native(
+        cls, native_cluster: gn.HierarchicalCluster
+    ) -> "HierarchicalCluster":
+        if not isinstance(native_cluster, gn.HierarchicalCluster):
+            raise TypeError(
+                "This class method is only valid for graspologic_native.HierarchicalCluster"
+            )
+        return cls(
+            node=native_cluster.node,
+            cluster=native_cluster.cluster,
+            parent_cluster=native_cluster.parent_cluster,
+            level=native_cluster.level,
+            is_final_cluster=native_cluster.is_final_cluster,
+        )
+
+    @staticmethod
+    def final_hierarchical_clustering(
+        hierarchical_clusters: List[
+            Union["HierarchicalCluster", gn.HierarchicalCluster]
+        ],
+    ) -> Dict[str, int]:
+        if not isinstance(hierarchical_clusters, list):
+            raise TypeError(
+                "This static method requires a list of hierarchical clusters"
+            )
+        final_clusters = (
+            cluster for cluster in hierarchical_clusters if cluster.is_final_cluster
+        )
+        return {cluster.node: cluster.cluster for cluster in final_clusters}
+
+
+def hierarchical_leiden(
+    graph: Union[List[Tuple[str, str, float]], nx.Graph],
+    max_cluster_size: int = 1000,
+    starting_communities: Optional[Dict[str, int]] = None,
+    iterations: int = 1,
+    resolution: float = 1.0,
+    randomness: float = 0.001,
+    use_modularity: bool = True,
+    random_seed: Optional[int] = None,
+    weight_attribute: str = "weight",
+    is_weighted: Optional[bool] = None,
+    weight_default: float = 1.0,
+    check_directed: bool = True,
+) -> List[HierarchicalCluster]:
+    """
+    Leiden is a global network partitioning algorithm. Given a list of edges and a maximization function, it will
+    iterate through the network attempting to find an optimal partitioning of the entire network. Unlike the function
+    graspologic.partition.leiden(), this function does not stop after maximization has been achieved. On some large
```

Comment:
```suggestion
    ``graspologic.partition.leiden()``, this function does not stop after maximization has been achieved. On some large
```
intersphinx would be even better

Reply:
Is intersphinx the nomenclature for "links within the docs"?  I presumed it was only for external and that we got this automatically.  I agree it should be a link directly to the function, though.